modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C

```diff
@@ -8,68 +8,42 @@
 
 template <>
 InputParameters
 validParams<PowerLawCreepStressUpdate>()
 {
-  InputParameters params = validParams<RadialReturnStressUpdate>();
-  params.addClassDescription("This class uses the discrete material in a radial return isotropic "
-                             "power law creep model.  This class can be used in conjunction with "
-                             "other creep and plasticity materials for more complex simulations.");
+  InputParameters params = validParams<RadialReturnCreepStressUpdateBase>();
+  params.addClassDescription(
+      "This class uses the stress update material in a radial return isotropic power law creep "
+      "model.  This class can be used in conjunction with other creep and plasticity materials for "
+      "more complex simulations.");
 
   // Linear strain hardening parameters
   params.addRequiredParam<Real>("coefficient", "Leading coefficent in power-law equation");
   params.addRequiredParam<Real>("n_exponent", "Exponent on effective stress in power-law equation");
   params.addParam<Real>("m_exponent", 0.0, "Exponent on time in power-law equation");
   params.addRequiredParam<Real>("activation_energy", "Activation energy");
   params.addParam<Real>("gas_constant", 8.3143, "Universal gas constant");
-  params.addParam<Real>("start_time", 0.0, "Start time (if not zero)");
-  params.addCoupledVar("temperature", 0.0, "Coupled temperature");
-  params.addDeprecatedParam<std::string>(
-      "creep_prepend",
-      "",
-      "String that is prepended to the creep_strain Material Property",
-      "This has been replaced by the 'base_name' parameter");
-  params.set<std::string>("effective_inelastic_strain_name") = "effective_creep_strain";
+  params.addDeprecatedParam<Real>("start_time",
+                                  0.0,
+                                  "Start time (if not zero).",
+                                  "This is automatically set from by the executioner start_time");
 
   return params;
 }
 
 PowerLawCreepStressUpdate::PowerLawCreepStressUpdate(const InputParameters & parameters)
-  : RadialReturnStressUpdate(parameters),
-    _creep_prepend(getParam<std::string>("creep_prepend")),
+  : RadialReturnCreepStressUpdateBase(parameters),
     _coefficient(getParam<Real>("coefficient")),
     _n_exponent(getParam<Real>("n_exponent")),
     _m_exponent(getParam<Real>("m_exponent")),
     _activation_energy(getParam<Real>("activation_energy")),
     _gas_constant(getParam<Real>("gas_constant")),
-    _start_time(getParam<Real>("start_time")),
-    _has_temp(isCoupled("temperature")),
-    _temperature(_has_temp ? coupledValue("temperature") : _zero),
-    _creep_strain(declareProperty<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain")),
-    _creep_strain_old(
-        getMaterialPropertyOld<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain"))
-{
-}
-
-void
-PowerLawCreepStressUpdate::initQpStatefulProperties()
+    _start_time(_app.getStartTime())
```

Comment:
The way you have it, `start_time` will be ignored if set by the user. This would use it if it was set, otherwise use the app's start time:
```
_start_time(parameters.isParamSetByUser("start_time") ? getParam<Real>("start_time") : _app.getStartTime())
```
The problem is that the default would no longer be zero, as is claimed, and as it used to be.

Reply:
Oh I see. This will retain functionality I suppose. Removing the default, I can then use `isParamValid()`. would that be better?

Reply:
I don't see a super clean way to avoid changing old behavior. Maybe do what I suggested above, and then just below in the body of the constructor, do this:
```
if (_m_exponent != 0.0 && _app.getStartTime() != 0.0 && !isParamSetByUser("start_time"))
  mooseError("The default start_time in PowerLawCreepStressUpdate is the executioner's start time, rather than 0");
```
Never mind -- that doesn't work either, because then it will never run if your executioner starts at a nonzero time.  Maybe just leave this whole start_time thing alone since it's unrelated to what you really want to do here?

Reply:
Checkout what I added. I think that gets the root of the reason it is there. If the user does tries to add their own start_time, it'll get a deprecated warning. Otherwise, it'll be set by the executioner's start time. I suppose I could throw a check to ensure that the user start time is greater than the executioner's start time. That'll take care of all cases.

Really, I want to just yank the input param as is...